src/Sylius/Bundle/ApiBundle/Doctrine/QueryExtension/CurrencyExtension.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\QueryExtension;
+
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final readonly class CurrencyExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
+{
+    public function __construct(private SectionProviderInterface $sectionProvider)
+    {
+    }
+
+    public function applyToCollection(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        Operation $operation = null,
+        array $context = [],
+    ): void {
+        if (!is_a($resourceClass, CurrencyInterface::class, true)) {
+            return;
+        }
+
+        if ($this->sectionProvider->getSection() instanceof AdminApiSection) {
```

Comment:
It's better to explicitly check for required value, than use value negation.
```suggestion
        if (!$this->sectionProvider->getSection() instanceof ShopApiSection) {
            return;
```

`ShopApiOrdersSubSection` extends `ShopApiSection`, so it should be alright

Reply:
I think it is a good idea, especially given the context of `ShopCurrencyExtension`